tests/test_meta/test_estimatortransformer.py

```diff
@@ -111,3 +113,27 @@ def test_kwargs(patched_clone, random_xy_dataset_clf):
     np.testing.assert_array_equal(
         sample_weights, estimator.fit.call_args[1]['sample_weight']
     )
+
+
+def test_nan_input():
+    """ Test X containing nan with check_X=False. """
+    X = np.array([[np.nan, 4], [7, 3], [5, 5], [7, 2], [5, 7]])
+    y = np.array([1, 0, 1, 0, 1])
+    clf = HistGradientBoostingClassifier()
+    transformer = EstimatorTransformer(clf, check_X=False)
+    transformed = transformer.fit(X, y).transform(X)
+
+    assert transformed.shape == (y.shape[0], 1)
+    assert np.all(transformed == clf.fit(X, y).predict(X))
+
+
+def test_object_input():
+    """ Test X containing string with check_X=False. """
+    X = np.array([["1", 4], ["7", 3], ["5", 5], ["7", 2], ["5", 7]])
+    y = np.array([1, 0, 1, 0, 1])
+    clf = HistGradientBoostingClassifier()
+    transformer = EstimatorTransformer(clf, check_X=False)
+    transformed = transformer.fit(X, y).transform(X)
+
+    assert transformed.shape == (y.shape[0], 1)
+    assert np.all(transformed == clf.fit(X, y).predict(X))
```

Comment:
It feels like these two tests are _very_ similar and might benefit from `pytest.parametrize`. If you appreciate an example, calmcode has a small tutorial. 

https://calmcode.io/pytest-tricks/parametrize-twice.html

You'd only need to parametrize once for this example, but this video has a nice example.

Reply:
thanks! I made it a parametrized test